### DISCUSSION TITLE
remote request to server from proxyclient.

### DISCUSSION THREAD

**Discussion by gonzalopixart** [justcoding121/Titanium-Web-Proxy#835] [General]
I have a function that sends a json to a server, this json has the url and the server to which I send this info analyzes it and returns a true or false, but it is not working for me. I save the return of this function in a variable that then executes the .OK with the blocking html code, I have attached an example of how I did it, but how would it be the best way to do it?
![image](https://user-images.githubusercontent.com/33877069/108554763-54204900-72d3-11eb-93e6-cb08e79cd121.png)

**Comment by honfika** [comment]
I think you should as k this on Stackoverflow, since this is not related to TWP.

**Reply by gonzalopixart** [reply]
I think that if, in a sector of the proxyserver of the solution for c # they do an analysis of the url and if it contains a certain string they let it pass or they block it. but they have it hardcoded

**Comment by justcoding121** [comment]
Best approach: keep `BeforeRequest` fully async and call your decision API with `HttpClient` (don't block with `.Result` / `.Wait()`).

```csharp
private static readonly HttpClient Http = new HttpClient();

private async Task OnRequest(object sender, SessionEventArgs e)
{
    var url = e.HttpClient.Request.Url;
    var payload = $"{{\"url\":\"{url}\"}}";

    using var content = new StringContent(payload, Encoding.UTF8, "application/json");
    using var response = await Http.PostAsync("https://your-server/check", content);
    var body = await response.Content.ReadAsStringAsync();

    var blocked = body.Contains("true"); // or deserialize properly
    if (blocked)
    {
        e.Ok("<html><body>Blocked</body></html>");
    }
}
```

Tips:
- Reuse one static `HttpClient`.
- Make sure your handler is `async Task` and registered as `proxyServer.BeforeRequest += OnRequest`.
- If the check fails/times out, decide a default (allow vs block) explicitly.

Closing ? this is mostly general async HTTP usage; reopen if Titanium specifically isn't firing `BeforeRequest`.
